internal/core/cloud/aws.go

```diff
@@ -151,11 +173,45 @@ func requestMetadata(ctx context.Context, client HTTPClient, url string) (interf
 		return nil, err
 	}
 
+	if resp.StatusCode == http.StatusUnauthorized || resp.StatusCode == http.StatusForbidden {
+		return nil, errors.Errorf("failed to fetch AWS metadata: %s", resp.Status)
+	}
+
 	// The metadata endpoint may return json elements
 	if json.Valid(body) {
-		var jsonData interface{}
+		var jsonData any
 		err := json.Unmarshal(body, &jsonData)
 		return jsonData, err
 	}
 	return string(body), nil
 }
+
+func requestMetadataToken(ctx context.Context, client HTTPClient) (string, error) {
```

Comment:
super nitpick: as this is the first function we use in `NewAWSMetadata`, could we put it just after it? to avoid that much scrolling

Reply:
Here it is 3cdd69d4a12a1305a7f8949908db31a354781128